tgui/packages/tgui/interfaces/LibraryConsole.js

```diff
@@ -0,0 +1,988 @@
+import { map, sortBy } from 'common/collections';
+import { flow } from 'common/fp';
+import { classes } from 'common/react';
+import { marked } from 'marked';
+import { useBackend, useLocalState } from '../backend';
+import { Box, Button, Dropdown, Input, Modal, NoticeBox, NumberInput, LabeledList, Section, Stack, Table } from '../components';
+import { Window } from '../layouts';
+import { sanitizeText } from "../sanitize";
+
+export const LibraryConsole = (props, context) => {
+  const { act, data } = useBackend(context);
+  const {
+    show_dropdown,
+    display_lore,
+  } = data;
+  return (
+    <Window
+      theme={display_lore ? "spookyconsole" : ""}
+      title="Library Terminal"
+      width={880}
+      height={520}>
+      <PopoutMenu />
+      <Box fillPositionedParent left={show_dropdown ? "164px" : "50px"}>
+        <Window.Content>
+          <PageDisplay pb="10px" />
+        </Window.Content>
+      </Box>
```

Comment:
Would it be easy for you to refactor this into stacks? Hard coded margins are bad.

Reply:
I don't really want to end up in a position where the text overflows the space allotted, or gets too small in comparison. Is that reasonable? or am I thinking about this wrong

Reply:
Yes, it's possible to do in a nicer way. But if it's too hard for you, then it's ok to keep this as is, just make sure the gap is 6px between menu bar and the content.

Reply:
![image](https://user-images.githubusercontent.com/58055496/157186963-6e24f8be-5d9e-46fc-b23c-2cc5fa2897d4.png)
What I want to do is something like this, which looks really nice, and dynamically accounts for string length, which is handy.
Trouble is what it does to modals. It makes them fill up to the whole window, rather then the tabs (for obvious reasons)
I can't nest the window content inside though, because if I do it just overflows without a hardcoded offset like that.

I'm partly doing this because I want to get better at js. I just don't know how I'd fix this without manually offsetting the window content, or not doing dimmers that way. Am happy to learn if it's possible, but I may be trying to do something that's inherently kinda crummy.

Regardless I'll ensure there's only a 6 px gap

Reply:
Style helped me with this. I love you style